### ISSUE TITLE
Does not support directory ending in '.elm'

### ISSUE THREAD

**Issue by dbandstra** [jfmengels/node-elm-review#52]
We have a convention of putting elm files in a folder called `src.elm` (to go alongside `src.js` for port code). But this doesn't appear to work with elm-review.

```
$ elm-review
-- NO FILES FOUND --------------------------------------------------------------

I was expecting to find Elm files in all the paths that you passed, but I could
not find any in the following directories:
- /home/db/app/src.elm

When I can’t find files in some of the directories, I’m assuming that you
misconfigured the CLI’s arguments.
```

I believe it's due to this special case: https://github.com/jfmengels/node-elm-review/blob/b3ae61b0e600fde3887ae344e2e0e2305ec73928/lib/elm-files.js#L215-L216

Maybe `if (path_.endsWith('.elm'))` should be replaced by checking if the path points to a regular file?

**Comment by jfmengels** [comment]
Hey @dbandstra :wave: 

Thanks for the issue! I am not able to reproduce the issue like you are: I am getting an issue that `elm-review` quits without a warning, but I can't reproduce what you are encoutering.
Would it be possible for you to create a small project structure (as a GitHub repo for instance) where you can reproduce the issue?

As far as the suggested solution goes, I guess that it would work but I'd like to prevent calling `fsStat` on every file, as that would make `elm-review` boot up slower.

I think this condition is only used when you pass in files/folders as arguments to the `elm-review` CLI, so maybe we could call `fsStat` only on those, and avoid it for the directories present in `source-directories`.

**Comment by dbandstra** [comment]
I think I must have got a little confused when reporting. You're right, the error only happens when explicitly specifying paths on the command line. If I just run `elm-review` by itself, it silently ignores source directories ending in `.elm` (and if I have multiple source directories but only some of them end in `.elm`, the ones that don't will still get reviewed).

From what I can tell, the line I highlighted is only run on each path passed on the command line, or each entry in `source-directories`. It's not recursing into the folders or anything. This seems fine to me? I do think we'd need to stat each one of them to fix the issue.

Here's an excerpt of getProjectFiles with comments added by me:

```js
function getProjectFiles() {
  ...

  // If options.directoriesToAnalyze is set, returns that list.
  // Otherwise, returns the "source-directories" (plus "tests").
  const sourceDirectories = getSourceDirectories(options, elmJson);

  // findFiles is the function I linked in my first post, that currently
  // assumes that if its argument ends in ".elm", it refers to a regular file.
  const filesInDirectories = await Promise.all(
    sourcesDirectories.map(findFiles)
  );

  const emptyDirectories = filesInDirectories.filter(
    ({files}) => files.length === 0
  );

  if (
    // The following line prevents elm-review from displaying
    // an error when source-directories contains only "src.elm".
    // Maybe this is not an issue if we fix the ".elm" problem.
    options.directoriesToAnalyze.length !== 0 &&
    emptyDirectories.length !== 0
  ) {
    throw new ErrorMessage.CustomError(
      'NO FILES FOUND',
```

**Comment by jfmengels** [comment]
Fix released in v2.5.4. Thanks for raising the issue and for looking into this! :pray: 
I hope you enjoy using `elm-review`!
